src/test/java/com/coherentsolutions/pot/insuranceservice/integration/controller/ClaimManagementControllerIt.java

```diff
@@ -0,0 +1,373 @@
+    Plan otherPlan = planRepository.save(
+        buildPlan("Vision Premium", dental, new BigDecimal("299.99")));
+    createClaimViaApi(
+        buildClaimDto(user.getId(), otherPlan.getId(), LocalDate.now(), new BigDecimal("15.00")));
+
+    // When / Then
+    mockMvc.perform(get(ENDPOINT)
+            .param("planName", "dental")
+            .param("page", "0").param("size", "10"))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.content.length()").value(1))
+        .andExpect(jsonPath("$.content[0].planName").value(plan.getName()));
+  }
+
+  @Test
+  @DisplayName("GET /v1/claims — filter by amountMin/amountMax (inclusive)")
+  void shouldFilterByAmountRangeInclusive() throws Exception {
+    // Given
+    createClaimViaApi(
+        buildClaimDto(user.getId(), plan.getId(), LocalDate.now(), new BigDecimal("10.00")));
+    ClaimDto mid = createClaimViaApi(
+        buildClaimDto(user.getId(), plan.getId(), LocalDate.now(), new BigDecimal("20.00")));
+    createClaimViaApi(
+        buildClaimDto(user.getId(), plan.getId(), LocalDate.now(), new BigDecimal("30.00")));
+
+    // When / Then
+    mockMvc.perform(get(ENDPOINT)
+            .param("amountMin", "20.00")
+            .param("amountMax", "20.00")
+            .param("page", "0").param("size", "10"))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.content.length()").value(1))
+        .andExpect(jsonPath("$.content[0].id").value(mid.getId().toString()));
+  }
+
+  private ClaimDto buildClaimDto(UUID userId, UUID planId, LocalDate serviceDate,
+      BigDecimal amount) {
+    return ClaimDto.builder()
+        .serviceDate(serviceDate)
+        .amount(amount)
+        .planId(planId)
+        .consumer(ConsumerDto.builder().userId(userId).build())
+        .build();
+  }
+
+  private ClaimDto createClaimViaApi(ClaimDto req) throws Exception {
+    String response = mockMvc.perform(
+            post(ENDPOINT).contentType(MediaType.APPLICATION_JSON)
+                .content(createClaimJson(
+                    req.getConsumer().getUserId(),
+                    req.getPlanId(),
+                    req.getServiceDate(),
+                    req.getAmount(),
+                    req.getStatus(),
+                    req.getClaimNumber()
+                )))
+        .andExpect(status().isCreated())
+        .andReturn().getResponse().getContentAsString();
+    return objectMapper.readValue(response, ClaimDto.class);
+  }
+
+  private Company createTestCompany(String name, String countryCode, String email, String website) {
+    Company company = new Company();
+    company.setName(name);
+    company.setCountryCode(countryCode);
+    company.setEmail(email);
+    company.setWebsite(website);
+    company.setStatus(CompanyStatus.ACTIVE);
+    return companyRepository.save(company);
+  }
+
+  private User createTestUser(String firstName, String lastName, String username, String email,
+      Company c, String ssn) {
+    User u = new User();
```

Comment:
Try to avoid using such one-letter names in code. Even if their meaning is clear.
```suggestion
    var user = new User();
```